### PR TITLE
Upgrade Ruma

### PR DESCRIPTION
**Pull request by jplatte**

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/olm/account.rs [hunk 1]
```diff
@@ -829,16 +832,15 @@ impl ReadOnlyAccount {
...
 
-        let mut signed_keys = BTreeMap::new();
-        signed_keys
-            .entry(self.user_id().to_owned())
-            .or_insert_with(BTreeMap::new)
-            .insert(public_key, serde_json::to_value(cross_signing_key)?);
+        let mut user_signed_keys = SignedKeys::new();
+        user_signed_keys.add_cross_signing_keys(public_key, Raw::new(&cross_signing_key)?);
+
+        let signed_keys = BTreeMap::from_iter([(self.user_id().to_owned(), user_signed_keys)]);
```
Comment (poljar) [inline crates/matrix-sdk-crypto/src/olm/account.rs:843]: A bunch of places already use this construct but use `BTreeMap::from()` instead. Let's stick to `from()` for consistency sake.

File: crates/matrix-sdk-crypto/src/olm/signing/mod.rs [hunk 2]
```diff
@@ -449,11 +452,10 @@ impl PrivateCrossSigningIdentity {
...
 
-        let mut signed_keys = BTreeMap::new();
-        signed_keys
-            .entry((*self.user_id).to_owned())
-            .or_insert_with(BTreeMap::new)
-            .insert(device_keys.device_id.to_string(), serde_json::to_value(device_keys)?);
+        let mut user_signed_keys = SignedKeys::new();
+        user_signed_keys.add_device_keys(device_keys.device_id.clone(), Raw::new(device_keys)?);
+
+        let signed_keys = BTreeMap::from_iter([((*self.user_id).to_owned(), user_signed_keys)]);
```
Comment (poljar) [inline crates/matrix-sdk-crypto/src/olm/signing/mod.rs:458]: Another one.

File: crates/matrix-sdk-crypto/src/olm/account.rs (hunk 1 above)
Comment (jplatte) [reply crates/matrix-sdk-crypto/src/olm/account.rs:843]: Oh, so we can actually just use `.into()` here because the type will be inferred from context.

**Comment by jplatte** [comment]
Rebased, pulled in two tiny refactorings from my xtask work and fixed some git history weirdness.

File: crates/matrix-sdk/src/room/common.rs [hunk 3]
```diff
@@ -7,10 +7,10 @@ use matrix_sdk_base::{
...
 use matrix_sdk_common::locks::Mutex;
 use ruma::{
-    api::client::r0::{
+    api::client::{
         filter::{LazyLoadOptions, RoomEventFilter},
         membership::{get_member_events, join_room_by_id, leave_room},
-        message::get_message_events::{self, Direction},
+        message::get_message_events::{self, v3::Direction},
         room::get_room_event,
         tag::{create_tag, delete_tag},
```
Comment (poljar) [inline crates/matrix-sdk/src/room/common.rs:15]: There's still a dead link for `delete_tag::Response` here.

**Review by poljar — APPROVED** [review]
Another dead link, feel free to merge when that's resolved.

**Pull request merged**